Kickstarter-iOS/Features/ChangeEmail/Controller/ChangeEmailView.swift

```diff
@@ -0,0 +1,195 @@
+
+        entryField(
+          titleText: Strings.Current_password(),
+          placeholderText: Strings.login_placeholder_password(),
+          secureField: true,
+          valueText: $passwordText
+        )
+        .currentPassword()
+        .focused($focusField, equals: .currentPassword)
+        .submitScope(passwordText.isEmpty)
+        .onSubmit {
+          focusField = nil
+        }
+
+        Color(.ksr_cell_separator).frame(maxWidth: .infinity, maxHeight: 1)
+      }
+      .listRowSeparator(.hidden)
+      .listRowInsets(EdgeInsets())
+    }
+    .navigationTitle(Strings.Change_email())
+    .background(Color(.ksr_support_100))
+    .listStyle(.plain)
+  }
+
+  @ViewBuilder
+  private func entryField(titleText: String,
+                          placeholderText: String,
+                          secureField: Bool,
+                          valueText: Binding<String>) -> some View {
+    HStack {
+      Text(titleText)
+        .frame(
+          maxWidth: .infinity,
+          alignment: .leading
+        )
+        .font(Font(UIFont.ksr_body()))
+        .foregroundColor(Color(.ksr_support_700))
+      Spacer()
+
+      newEntryField(secureField: secureField, placeholderText: placeholderText, valueText: valueText)
+    }
+    .padding(12)
+    .accessibilityElement(children: .combine)
+    .accessibilityLabel(titleText)
+  }
+
+  @ViewBuilder
+  private func newEntryField(secureField: Bool,
+                             placeholderText: String,
+                             valueText: Binding<String>) -> some View {
+    if secureField {
+      SecureField(
+        "",
+        text: valueText,
+        prompt: Text(placeholderText).foregroundColor(Color(.ksr_support_400))
+      )
+    } else {
+      TextField(
+        "",
+        text: valueText,
+        prompt:
+        Text(placeholderText).foregroundColor(Color(.ksr_support_400))
+      )
+    }
+  }
+}
+
+@available(iOS 15.0, *)
+struct EntryFieldModifier: ViewModifier {
+  let keyboardType: UIKeyboardType
+  let textColor: Color
+  let submitLabel: SubmitLabel
+  let editable: Bool
+  let titleText: String
+
+  func body(content: Content) -> some View {
+    content
+      .frame(
+        maxWidth: .infinity,
+        alignment: .trailing
+      )
+      .keyboardType(self.keyboardType)
+      .font(Font(UIFont.ksr_body()))
+      .foregroundColor(self.textColor)
+      .lineLimit(1)
+      .multilineTextAlignment(.trailing)
+      .submitLabel(self.submitLabel)
+      .disabled(!self.editable)
+      .accessibilityElement()
+      .accessibilityLabel(self.titleText)
+  }
+}
+
+@available(iOS 15.0, *)
+extension View {
+  func currentEmail(keyboardType: UIKeyboardType = .default,
+                    textColor: Color = Color(.ksr_support_700),
+                    submitLabel: SubmitLabel = .return,
+                    editable: Bool = false,
+                    titleText: String = Strings.Current_email()) -> some View {
+    modifier(EntryFieldModifier(
```

Comment:
Is there a difference between using a ViewModifier here instead of a ViewBuilder?

Also, maybe we can extract this out into a more reusable component since the text fields on change email and change password screens are so similar. What do you think?

Reply:
will address in this [PR](https://github.com/kickstarter/ios-oss/pull/1803)